src/selectors/entities/typing.js

```diff
@@ -11,25 +11,36 @@ import {getTeammateNameDisplaySetting} from 'selectors/entities/preferences';
 
 import {displayUsername} from 'utils/user_utils';
 
+const getUsersTypingImpl = (profiles, teammateNameDisplay, channelId, parentPostId, typing) => {
+    const id = channelId + parentPostId;
+
+    if (typing[id]) {
+        const users = Object.keys(typing[id]);
+
+        if (users.length) {
+            return users.map((userId) => {
+                return displayUsername(profiles[userId], teammateNameDisplay);
+            });
+        }
+    }
+
+    return [];
+};
+
+export const getUsersTypingByChannelAndPost = createSelector(
```

Comment:
This should be done as `makeGetUsersTyping` or something along those lines where it returns a selector. Otherwise, you'll be calling `getUsersTypingByChannelAndPost` for both the center channel and RHS with different arguments which will prevent it from being memoized

Reply:
Very good catch, thanks! :)